src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,175 @@
+        user1.setPassword(" ");
+        user1.setAge(18);
+        assertThrows(AuthenticationException.class, () -> {
+            registrationService.register(user1);
+        }, "Password less than 6 characters");
+        user1.setPassword("$$$$");
+        assertThrows(AuthenticationException.class, () -> {
+            registrationService.register(user1);
+        }, "Password less than 6 characters");
+        user1.setPassword("///");
+        assertThrows(AuthenticationException.class, () -> {
+            registrationService.register(user1);
+        }, "Password less than 6 characters");
+        user1.setPassword(".");
+        assertThrows(AuthenticationException.class, () -> {
+            registrationService.register(user1);
+        }, "Password less than 6 characters");
+        user1.setPassword(" ");
+        assertThrows(AuthenticationException.class, () -> {
+            registrationService.register(user1);
+        }, "Password less than 6 characters");
+    }
+
+    @Test
+    void userAge_notOk() {
+        User user1 = new User();
+        user1.setLogin("SuperUser");
+        user1.setPassword("111111");
+        user1.setAge(17);
+        assertThrows(AuthenticationException.class, () -> {
+            registrationService.register(user1);
+        }, "Age less than 18 years");
+        user1.setAge(0);
+        assertThrows(AuthenticationException.class, () -> {
+            registrationService.register(user1);
+        }, "Age less than 18 years");
+        user1.setAge(-20);
+        assertThrows(AuthenticationException.class, () -> {
+            registrationService.register(user1);
+        }, "Age less than 18 years");
+    }
+
+    @Test
+    void nullLogin_notOk() {
+        User user1 = new User();
+        user1.setLogin(null);
+        user1.setPassword("111111");
+        user1.setAge(17);
+        assertThrows(NullPointerException.class, () -> {
+            registrationService.register(user1);
+        });
+    }
+
+    @Test
+    void nullPassword_notOk() {
+        User user1 = new User();
+        user1.setLogin("UserSup");
+        user1.setPassword(null);
+        user1.setAge(17);
+        assertThrows(NullPointerException.class, () -> {
+            registrationService.register(user1);
+        });
+    }
+
+    @Test
+    void nullAge_notOk() {
+        User user1 = new User();
+        user1.setLogin("UserSup");
+        user1.setPassword("123456");
+        user1.setAge(null);
+        assertThrows(NullPointerException.class, () -> {
+            registrationService.register(user1);
+        });
+    }
+
+    @Test
+    void nullUser_notOk() {
+        User user1 = null;
+        assertThrows(NullPointerException.class, () -> {
+            registrationService.register(user1);
+        });
+    }
+
+    @Test
+    void maxValueIntAge_notOk() {
+        User user1 = new User();
+        user1.setLogin("UserSup");
+        user1.setPassword("123456");
+        user1.setAge(Integer.MAX_VALUE + 1);
```

Comment:
do you know that Integer.MAX_VALUE + 1 == Integer.MIN_VALUE ? :) better remove this test, you've tested age already